### ISSUE TITLE
[DOXIASITETOOLS-1] Provide a schema for site.xml

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia-sitetools#215]
**[Wendy Smoak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=wsmoak)** opened **[DOXIASITETOOLS-1](https://issues.apache.org/jira/browse/DOXIASITETOOLS-1?redirect=false)** and commented

Brett mentioned that a schema for site.xml will be available with maven-site-plugin 2.0 final.  Just logging a reminder issue as I don't see it in svn or at http://maven.apache.org/xsd/ .

http://mail-archives.apache.org/mod_mbox/maven-users/200602.mbox/%3c9e3862d80602031847h54438b4ft42e61f5481a3f34f@mail.gmail.com%3e

---

**Affects:** 1.0-alpha-9

**Issue Links:**
- [MSITE-320](https://issues.apache.org/jira/browse/MSITE-320) Reference documentation

20 votes, 15 watchers

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Modello has some bugs so we need to wait for a new release before.

http://www.nabble.com/Provide-site-descriptor--%28was%3A-Missing-site-descriptor%29-tf2042354s177.html#a5622466

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

So, if I understand you correctly Vincent, this issue is in reality a doxia issue. But it is held back because of bugs or missing features in modello?

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

The decoration model is a Doxia feature so this issue could be probably in Doxia.
Denis Cabasson fixed small bugs in Modello (like list rendering). IMHO we need to apply these before to deploy decoration descriptor.

**Comment by jira-importer** [comment]
**[Lukas Theussl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ltheussl)** commented

Dennis, Vincent,

can we close this now with the current decoration-model? The xsd is already at http://maven.apache.org/xsd/decoration-1.0.0.xsd .

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

I didn't know we had already published the xsd. I made a diff between the "published" decoration model and one generated from the current 1.0.x branch. Here are the results:

```
diff published-decoration-1.0.0.xsd generated-decoration-1.0.0.xsd
62c62
<             <xs:element name="logo" minOccurs="0" maxOccurs="unbounded" type="Logo"/>
&mdash;
>             <xs:element name="poweredBy" minOccurs="0" maxOccurs="unbounded" type="Logo"/>
82c82
<       <xs:element minOccurs="0" name="custom">
&mdash;
>       <xs:element minOccurs="0">
155c155
<       <xs:element minOccurs="0" name="head">
&mdash;
>       <xs:element minOccurs="0">
177c177
<             <xs:element name="item" minOccurs="0" maxOccurs="unbounded" type="LinkItem"/>
&mdash;
>             <xs:element name="link" minOccurs="0" maxOccurs="unbounded" type="LinkItem"/>
190c190
<             <xs:element name="item" minOccurs="0" maxOccurs="unbounded" type="LinkItem"/>
&mdash;
>             <xs:element name="breadcrumb" minOccurs="0" maxOccurs="unbounded" type="LinkItem"/>
```

This puts us in a difficult position. Have we officially published the xsd? If so, then we must probably change the model version to 1.0.1 or something. That kind of sucks, since it is version 1.0.0 of the product itself. What do you guys think?

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Sounds like Modello differences. I guess the differences come that no version of modello-maven-plugin is specify [1] or its parent. Try to remake diff with Modello 1.0 in your local repo.

[1] https://svn.apache.org/repos/asf/maven/doxia/doxia-sitetools/branches/doxia-sitetools-1.0.x/doxia-decoration-model/pom.xml
